nrf_modem/include/nrf_modem.h

```diff
@@ -171,9 +171,27 @@ void nrf_modem_application_irq_handler(void);
  * @brief Trace IRQ handler in the modem library.
  *
  * Call this function when handling the Trace IRQ.
+ *
  */
 void nrf_modem_trace_irq_handler(void);
 
+/**
+ * @brief Function to indicate that the application has completed the processing of a trace buffer
+ *
+ * The application shall call this function to let the modem library free the trace memory
+ * pointed to by @p buf. It is the application's responsibility to call this function with
+ * the same parameter values as received in the @ref nrf_modem_os_trace_put function.
+ * Calling this function with incorrect values leads to undefined behavior.
+ *
+ * @param buf Pointer to the memory buffer as received in @ref nrf_modem_os_trace_put
+ * @param len Length of memory buffer as received in @ref nrf_modem_os_trace_put
+ * @retval Zero on success.
+ * @retval -NRF_EINVAL @p buf is @c NULL or an invalid trace buffer
+ * @retval -NRF_EINVAL @p len is too large to be a valid trace length
+ * @retval -NRF_EAGAIN Resource temporarily unavailable. Try again with same parameters.
```

Comment:
Try again when, is there some indicator of when to retry?

Reply:
@balaji-nordic

Reply:
There is no indicator. It should be treated similar to a `k_sem_take()` failure with `-EAGAIN` return.